Automattic/wp-calypso
# Migrate Titan mailbox set up form to new mailbox list component

#### Overview 

The next set of changes 

#### Proposed Changes

* Migrates instances of the provider specific `<TitanNewMailboxList/>` to the generic `<NewMailboxList/>` for the Titan mailbox setup form. This is the form used for creating already purchased mailboxes for users; there's no need for checkout.
* For the main view involved, the JSX components are converted to their TSX counterparts
* Refactors the `useCreateTitanMailboxMutation()` hook such that the internal mutation now accepts variables in the late stage using the `mutateAsync()` function
* Installs autoFocus feature

#### Testing Instructions



* Have a Professional Email subscription with multiple mailboxes ready
* Checkout this branch in your local development setup, or use [Calypso-live ](https://calypso.live/?image=registry.a8c.com/calypso/app:build-35778)
* Navigate to the Email plan page for the subscription i.e. [ **Upgrades** ] > [ **Domains** ] > [ **Active domain** (i.e. example.com) ]  > [ **(\d+) mailboxes** ]
* Open another browser tab to the email provider's external control panel. You can reach that from the Email plan page, using the [ **Manage all mailboxes** ] nav item. Use any of the external nav items exposed to access the external control panel
* Remove one or more mailboxes from the subscription using the external control panel
* Back at the open tab where the Email plan page is open, confirm that the deletion change is reflected ( i.e. one less mailboxes are now displayed)
* Confirm that an `Action required` notice with a CTA to `Set up mailbox` is now visible
*  Use the CTA to reveal the new mailbox form
* Confirm that form's behaviour is identical to the former behaviour before checkout. You can establish this by opening up another tab with the same form using the production website i.e. `https://wordpress.com/***`
* Confirm that you can create a new mailbox for the subscription without going through checkout

| Illustration             |
:------------------:
<img width="600" alt="Screenshot 2022-06-18 at 4 25 09 AM" src="https://user-images.githubusercontent.com/277661/174421382-5f76d428-1e28-48a0-ae4d-68f217be8fe8.png"> |
<img width="600" alt="Screenshot 2022-06-18 at 4 25 30 AM" src="https://user-images.githubusercontent.com/277661/174421389-ad2d3181-754d-4bb0-92b8-314d11b06f4b.png"> |
<img width="600" alt="Screenshot 2022-06-18 at 4 26 17 AM" src="https://user-images.githubusercontent.com/277661/174421401-bc88a72f-6538-4287-afed-e1eddf202202.png"> |
<img width="600" alt="Screenshot 2022-06-18 at 4 26 43 AM" src="https://user-images.githubusercontent.com/277661/174421405-f7873c9b-0a27-40c7-8c05-12bce2373835.png"> |
<img width="600" alt="Screenshot 2022-06-18 at 4 38 18 AM" src="https://user-images.githubusercontent.com/277661/174421453-5694df2d-23ee-4b88-9a60-d508c3b7f94f.png"> |



Related to #64424
